### PR TITLE
Update room field

### LINKED ISSUE #165: [PE-D][Tester B] Inputting in room numbers of 1 character aren't allowed
Applies to both `add` and `edit`

`edit 1 rm/1` does not work:

`edit 1 rm/g` does not work either:

`add n/John Doe p/98765432 e/johnd@example.com rm/5 dr/13/09/22 - 15/09/22 ng/3 rq/Extra Towels` does not work:

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: NicholasTYD/ped#4

### LINKED ISSUE #171: [PE-D][Tester A] Put a limit on the length of characters for rooms
Not really a bug but a suggestion. I guess this is just a user being dumb but you can consider doing this.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: KeithPJX/ped#5

### PR DESCRIPTION
**Pull request by kangzongxian**
Limit length to up to 100 characters
Fix regex so that 1 character only can be entered as well

close #171 and close #165

### CODE REVIEW

File: src/main/java/seedu/guest/model/guest/Room.java [hunk 1]
```diff
@@ -12,12 +12,16 @@ public class Room {
     public static final String MESSAGE_CONSTRAINTS =
             "Rooms should only contain alphanumeric characters and hyphens (-). "
             + "They cannot start or end with a hyphen. "
-            + "Hyphens can only be used between 2 alphanumeric characters.";
+            + "Hyphens can only be used between 2 alphanumeric characters."
+            + "You should enter a minimum of 1 character and a maximum of 100 characters.";
```
Comment (JustinPeng13) [inline src/main/java/seedu/guest/model/guest/Room.java]: Could we use passive voice instead? "They must be at least 1 character and at most 100 characters long." And insert this line before "They cannot start or end with a hyphen." as the length constraint is more important

File: src/test/java/seedu/guest/model/guest/RoomTest.java [hunk 2]
```diff
@@ -25,20 +25,24 @@ public void isValidRoom() {
...
+        assertFalse(Room.isValidRoom("A".repeat(101))); // above 100 characters is not allowed (near boundary values)
 
         // valid room
         assertTrue(Room.isValidRoom("05-73")); // only alphanumeric characters and hyphen
         assertTrue(Room.isValidRoom("RoomA")); // only alphanumeric characters
+        assertTrue(Room.isValidRoom("1")); // 1 character is allowed (near boundary values)
         assertTrue(Room.isValidRoom("Room5")); // hyphens are optional
         assertTrue(Room.isValidRoom("Aloha-5")); // only alphanumeric characters and hyphen
+        assertTrue(Room.isValidRoom("A".repeat(100))); // 100 characters is allowed (near boundary values)
+        assertTrue(Room.isValidRoom("A".repeat(80))); // less than 100 characters is allowed
```
Comment (JustinPeng13) [inline src/test/java/seedu/guest/model/guest/RoomTest.java]: Don't think we need this as we have test cases with less than 100 characters already!

File: src/test/java/seedu/guest/model/guest/RoomTest.java [hunk 3]
```diff
@@ -25,20 +25,24 @@ public void isValidRoom() {
         assertThrows(NullPointerException.class, () -> Room.isValidRoom(null));
 
         // invalid room
-        assertFalse(Room.isValidRoom("")); // empty string
-        assertFalse(Room.isValidRoom(" ")); // spaces only
+        assertFalse(Room.isValidRoom("")); // empty string, near boundary values
+        assertFalse(Room.isValidRoom(" ")); // spaces only, near boundary values
         assertFalse(Room.isValidRoom("-50-23")); // not allowed to start with a hyphen
```
Comment (JustinPeng13) [inline src/test/java/seedu/guest/model/guest/RoomTest.java:30]: Can we check not allowed to end with hyphen as well?

File: src/test/java/seedu/guest/model/guest/RoomTest.java [hunk 4]
```diff
@@ -25,20 +25,24 @@ public void isValidRoom() {
...
         assertFalse(Room.isValidRoom("50 - 23")); // not allowed to have spacings in between
```
Comment (JustinPeng13) [inline src/test/java/seedu/guest/model/guest/RoomTest.java]: Can we change this to only 1 space to be closer to the boundary of "between 2 alphanumeric characters"?

File: src/test/java/seedu/guest/model/guest/RoomTest.java [hunk 5]
```diff
@@ -25,20 +25,24 @@ public void isValidRoom() {
...
         assertFalse(Room.isValidRoom("50--23")); // not allowed to have more than 1 hyphen in between
         assertFalse(Room.isValidRoom("$50")); // '$' is not allowed
         assertFalse(Room.isValidRoom("^&*")); // special symbols only are not allowed
         assertFalse(Room.isValidRoom("aloha-5()")); // '(' or ')' is not allowed
```
Comment (JustinPeng13) [inline src/test/java/seedu/guest/model/guest/RoomTest.java]: Not sure if all 3 test cases are needed to test symbols. Maybe just have 1 test case like "5(1)A"? What do you think?

**Review by JustinPeng13 — CHANGES_REQUESTED** [review]

File: src/test/java/seedu/guest/model/guest/RoomTest.java (hunk 2 above)
Comment (kangzongxian) [reply src/test/java/seedu/guest/model/guest/RoomTest.java]: Sounds good!

File: src/main/java/seedu/guest/model/guest/Room.java (hunk 1 above)
Comment (kangzongxian) [reply src/main/java/seedu/guest/model/guest/Room.java]: Agreed!

File: src/test/java/seedu/guest/model/guest/RoomTest.java (hunk 3 above)
Comment (kangzongxian) [reply src/test/java/seedu/guest/model/guest/RoomTest.java:30]: Agreed!

File: src/test/java/seedu/guest/model/guest/RoomTest.java (hunk 4 above)
Comment (kangzongxian) [reply src/test/java/seedu/guest/model/guest/RoomTest.java]: Agreed!

File: src/test/java/seedu/guest/model/guest/RoomTest.java (hunk 5 above)
Comment (kangzongxian) [reply src/test/java/seedu/guest/model/guest/RoomTest.java]: Sounds good!

**Review by JustinPeng13 — APPROVED** [review]
